src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,71 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { resetCurrentTodo, selectCurrentTodo } from '../../features';
 
 export const TodoModal: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [todoUser, setTodoUser] = useState<User | null>(null);
+  const dispatch = useAppDispatch();
+
+  const activeTodo = useAppSelector(selectCurrentTodo);
+
+  useEffect(() => {
+    if (activeTodo) {
+      getUser(activeTodo.userId)
+        .then(setTodoUser)
+        .finally(() => setIsLoading(false));
+    }
+  }, [activeTodo]);
+
```

Comment:
The `useEffect` dependency array should include all variables used inside the effect that are not constants or functions defined outside the component. Ensure that `getUser` is a stable function or include it in the dependency array if it can change.